runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -93,39 +103,115 @@ public void mergeInto(MultiObjectSMREntry other) {
+                        .publishPercentiles(TRACKED_PERCENTILES)
+                        .publishPercentileHistogram(true)
+                        .register(registry));
         int numStreams = b.readInt();
-        for (int i = 0; i < numStreams; i++) {
-            UUID streamId = new UUID(b.readLong(), b.readLong());
-
-            // The MultiObjectSMREntry payload is structure as follows:
-            // LogEntry Type | number of MultiSMREntry entries | MultiSMREntry id | serialized MultiSMREntry | ...
-            // Therefore we need to unpack the MultiSMREntry entries one-by-one
-            int start = b.readerIndex();
-            MultiSMREntry.seekToEnd(b);
-            int multiSMRLen = b.readerIndex() - start;
-            b.readerIndex(start);
-            byte[] streamUpdates = new byte[multiSMRLen];
-            b.readBytes(streamUpdates);
-            streamBuffers.put(streamId, streamUpdates);
-
+        try {
+            super.deserializeBuffer(b, rt);
+            for (int i = 0; i < numStreams; i++) {
+                UUID streamId = new UUID(b.readLong(), b.readLong());
+                Optional<Timer> deserializeStreamTimer = MeterRegistryProvider.getInstance()
+                        .map(registry -> Timer.builder(METRIC_PREFIX + "." + "deserialize.stream")
+                                .tags(STREAM_ID, streamId.toString())
+                                .publishPercentiles(TRACKED_PERCENTILES)
+                                .publishPercentileHistogram(true)
+                                .register(registry));
+                Optional<Timer.Sample> deserializeStreamSample = MeterRegistryProvider.getInstance()
+                        .map(Timer::start);
+                Optional<DistributionSummary> streamDistributionSummary = MeterRegistryProvider.getInstance().map(registry ->
+                        DistributionSummary.builder(METRIC_PREFIX + "." + "deserialize.stream.size")
+                                .tags(STREAM_ID, streamId.toString())
+                                .publishPercentiles(TRACKED_PERCENTILES)
+                                .publishPercentileHistogram(true)
+                                .register(registry));
+                // The MultiObjectSMREntry payload is structure as follows:
+                // LogEntry Type | number of MultiSMREntry entries | MultiSMREntry id | serialized MultiSMREntry | ...
+                // Therefore we need to unpack the MultiSMREntry entries one-by-one
+                int multiSMRLen = 0;
+                try {
+                    int start = b.readerIndex();
+                    MultiSMREntry.seekToEnd(b);
+                    multiSMRLen = b.readerIndex() - start;
+                    b.readerIndex(start);
+                    byte[] streamUpdates = new byte[multiSMRLen];
+                    b.readBytes(streamUpdates);
+                    streamBuffers.put(streamId, streamUpdates);
+                } finally {
```

Comment:
Why do you need a finally block? If there is a failure you don't want to add that data point. 

If it fails the variable will go out of scope and get GC'd.